### DISCUSSION TITLE
error while deploying the SimpleStorage contract using foundry scripts

### DISCUSSION THREAD

**Discussion by johan784** [Cyfrin/foundry-full-course-cu#2432] [Q&A] [answered]
i am facing the following error while deplying my contract 

' ' '
forge script script/DeploySimpleStorage.s.
sol
[⠊] Compiling...
No files changed, compilation skipped
Error: 
Could not find target contract
' ' '

this is my code 

' ' '
import {Script} from "forge-std/Script.sol";
import{SimpleStorage} from  "../src/SimpleStorage.sol";

contract DeploySimpleStorage  is Script{

    function run() external returns (SimpleStorage) {
        vm.startBroadcast();

        SimpleStorage simplestorage= new SimpleStorage();
    

    vm.stopBroadcast();
    return simplestorage;
' ' '

**Comment by EngrPips** [comment]
Hello @johan784, I think the command is supposed to be the below. 

```js
forge script script/DeploySimpleStorage.s.sol:DeploySimpleStorage
```

**Reply by johan784** [reply]
it still says displays an error that it " Can't find a target contract"

**Reply by EngrPips** [reply]
Please show your working directory like i show mine below. Please open the script folder.
<img width="1440" alt="Screenshot 2024-09-09 at 15 02 25" src="https://github.com/user-attachments/assets/be9a3a54-9131-41e2-9643-232f42ef1dd2">

**Reply by johan784** [reply]
![image](https://github.com/user-attachments/assets/d7b1bb95-3259-4999-8fd0-aabbef92be04)

**Reply by EngrPips** [reply]
Open `DeploySimpleStorage.s.sol` file and show it.

**Reply by johan784** [reply]
<img width="1064" alt="Screenshot 2024-09-10 at 11 43 27 AM" src="https://github.com/user-attachments/assets/de084c3f-c84b-46e4-b3a7-2076bed44472">

**Reply by EngrPips** [reply, marked as answer]
Everything seems correct. Please run `foundryup`, then `forge clean`, then retry.

**Reply by johan784** [reply]
it worked!! Thank you so much!!

**Reply by EngrPips** [reply]
You are welcome. Keep crushing it.
